vault-client/init.sls

```diff
@@ -0,0 +1,26 @@
+{%- if 'vault' in pillar %}
```

Comment:
Remove this outer `if`, we need to provide reasonable defaults, or if there are required settings then reference them directly from the dictionary so they fail if not present. If the user calls this module without configuring the required pillar, that's a user error and it should fail.